packages/app/src/cli/services/dev.ts

```diff
@@ -280,12 +282,24 @@ async function dev(options: DevOptions) {
 
   await logMetadataForDev({devOptions: options, tunnelUrl: frontendUrl, shouldUpdateURLs, storeFqdn})
 
-  await reportAnalyticsEvent({config: options.commandConfig})
+  const abortController = new AbortController()
+  abortController.signal.addEventListener('abort', async () => {
+    setInterval(() => {
```

Comment:
1. Why `setInterval` not `setTimeout`? Either way it'll only run once because it kills the process, but I think `setTimeout` is more semantically correct.
2. Shouldn't we capture the timeout and cancel it if we shut down gracefully? And if we will always run the function because we can't be sure, we shouldn't display a message indicating that "graceful shutdown failed"

Reply:
> Why setInterval not setTimeout? Either way it'll only run once because it kills the process, but I think setTimeout is more semantically correct.

👍

> Shouldn't we capture the timeout and cancel it if we shut down gracefully? And if we will always run the function because we can't be sure, we shouldn't display a message indicating that "graceful shutdown failed"

yep, we should cancel it otherwise the `graceful shutdown` will never finish in less than the `GRACEFUL_EXIT_TIMEOUT` 
Regarding not adding `graceful shutdown failed` we could add it in the `--verbose` mode, just to have the chance to check it in case we need to (although it's true that we will see the exit(1)`

Reply:
Disable `dev preview` has been removed from this PR, so everything related with using `abortController` vs `treeKill` does not apply anymore